heron/common/src/cpp/network/network_error.h

```diff
@@ -30,4 +33,12 @@ enum NetworkErrorCode {
   TIMEOUT
 };
 
+static inline std::ostream& operator<<(std::ostream& os, const NetworkErrorCode code) {
+  std::string strs[] = { "OK", "Connection error", "Read error", "Write error",
```

Comment:
There is always some way to do it. Again, why do you think it is worth our effort to do it? For me, we only have one case so far, no need to bother a complex way to make it unnecessarily elegant. If you believe it is worth, you are welcome to do it in a separate PR.